cpp/src/arrow/compute/kernels/scalar_nested_test.cc

```diff
@@ -39,6 +39,46 @@ TEST(TestScalarNested, ListValueLength) {
   }
 }
 
+TEST(TestScalarNested, ListElement) {
+  for (auto ty : {list(int16()), fixed_size_list(uint8(), 2)}) {
+    auto input = ArrayFromJSON(ty, "[[8, 5], [74, 92], [7, 4]]");
+    auto out_ty = ty->id() == Type::LIST ? int16() : uint8();
```

Comment:
Why not just keep the element type consistent?

Reply:
To have more coverage in the test.

Reply:
In that case (and for the one above) I think it'd make more sense to test one group with list and large_list, where the list lengths are all different (and with null lists), and test fixed_size_list separately. And you can cast `ty` to `BaseListType` to use `value_type()` for the output type.

Reply:
Thanks!